auth-starter/src/main/java/com/vaadin/auth/starter/VaadinAuthContextImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.vaadin.auth.starter;
+
+import com.vaadin.flow.server.VaadinServletRequest;
+import com.vaadin.flow.server.VaadinServletResponse;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.security.web.authentication.logout.CompositeLogoutHandler;
+import org.springframework.security.web.authentication.logout.LogoutSuccessEventPublishingLogoutHandler;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.Optional;
+
+public class VaadinAuthContextImpl implements VaadinAuthContext {
+
+    private final VaadinAuthProperties properties;
+
+    private final ApplicationEventPublisher eventPublisher;
+
+    private final ClientRegistrationRepository clientRegistrationRepository;
+
+    /**
+     * Default implementation of the {@link VaadinAuthContext} interface.
+     */
+    public VaadinAuthContextImpl(VaadinAuthProperties properties,
+            ApplicationEventPublisher eventPublisher,
+            ClientRegistrationRepository clientRegistrationRepository) {
+        this.properties = properties;
+        this.eventPublisher = eventPublisher;
+        this.clientRegistrationRepository = clientRegistrationRepository;
+    }
+
+    @Override
+    public Optional<OidcUser> getAuthenticatedUser() {
+        return Optional.of(SecurityContextHolder.getContext())
+                .map(SecurityContext::getAuthentication)
+                .map(Authentication::getPrincipal)
+                .filter(OidcUser.class::isInstance).map(OidcUser.class::cast);
+    }
+
+    @Override
+    public void logout() {
```

Comment:
Does this solution tackle the problem described by @heruan `But this does not take into account any custom logout handler that could be added to the security configuration, so it's not optimal.`? If not, it probably has to be documented and if feasible Vaadin should provide it's own hooking mechanism to allow people to hook into the logout

Reply:
At this point it does not, but we already have prototyped a proper alternative that will be replaced before this gets merged: all the handlers (Spring and any custom one) will be collected right before the security chain is built, so they'll be used in this method.